### PR TITLE
Add nice stuff to 4.3 stable

### PR DESCRIPTION
**Pull request by epsilon-phase**
This contains:

* The original configuration reload work
* The `get_script_output_single_line` rewrite
* The `irods-grid reload` command

### CODE REVIEW

File: scripts/irods/test/test_configuration_reload.py [hunk 1]
```diff
@@ -97,18 +98,20 @@ def test_reload_command__issue_6529(self):
...
-            obj = json.loads(out)
-            changes = obj["hosts"][0]["configuration_changes_made"]
-
-            # And check the properties such that they are what we expect
-            self.assertTrue(len(changes) == 1)
-            self.assertTrue(changes[0]["path"] == f"/{PROP_NAME}")
-            self.assertTrue(changes[0]["op"] == "add")
-            self.assertTrue(changes[0]["value"] == "Yes?")
+            with session.make_session_for_existing_admin() as admin_session:
+                admin_session.environment_file_contents = IrodsConfig().client_environment
```
Comment (alanking) [inline scripts/irods/test/test_configuration_reload.py:102]: Still not sure that I understand the reason for this. Does this not work if we don't load the `IrodsConfig().client_environment` into the `admin_session.environment_file_contents`?

File: scripts/irods/test/test_configuration_reload.py [hunk 2]
```diff
@@ -97,18 +98,20 @@ def test_reload_command__issue_6529(self):
...
+
+                # Load the output of the command
+                obj = json.loads(out)
+                changes = obj["hosts"][0]["configuration_changes_made"]
+
+                # And check the properties such that they are what we expect
+                self.assertTrue(len(changes) == 1)
+                self.assertTrue(changes[0]["path"] == f"/{PROP_NAME}")
+                self.assertTrue(changes[0]["op"] == "add")
+                self.assertTrue(changes[0]["value"] == "Yes?")
```
Comment (alanking) [inline scripts/irods/test/test_configuration_reload.py:114]: Since we're in here touching this again, these could be `assertEqual`s.

**Review by alanking — COMMENTED** [review]
Cherry-picks look clean to me after eyeballing a diff with main. Had a couple comments on the test thing. Good catch on that.

File: scripts/irods/test/test_configuration_reload.py (hunk 1 above)
Comment (epsilon-phase) [reply scripts/irods/test/test_configuration_reload.py:102]: Yeah, irods-grid doesn't work without this line

**Review by alanking — APPROVED** [review]

**Pull request merged**